src/subscript/check_swatinit/check_swatinit.py

```diff
@@ -0,0 +1,612 @@
+        else:
+            del grid_df["SWATINIT_DECK"]  # This is not needed
+
+    # PCW from INIT is:
+    # "Output of scaled maximum water capillary pressures (provided
+    # data is supplied via the PCW keyword), or the scaling factor for
+    # the JFUNC keyword"
+
+    # Exposed to issues with endpoint scaling in peculiar decks:
+    satfunc_df = ecl2df.satfunc.df(eclfiles)
+
+    # Merge in the input pcmax pr. satnum for each cell:
+    grid_df = merge_pc_max(grid_df, satfunc_df)
+
+    grid_df = merge_equil(grid_df, ecl2df.equil.df(eclfiles, keywords=["EQUIL"]))
+
+    if "PPCWMAX" in eclfiles.get_ecldeck():
+        grid_df["PPCWMAX"] = ppcwmax_gridvector(eclfiles)
+
+    grid_df = augment_grid_frame_qc_vectors(grid_df)
+
+    if "SWATINIT" in grid_df:
+        grid_df["QC_FLAG"] = qc_flag(grid_df)
+
+    # Above the gas-oil contact, the computed capillary pressure will
+    # be p_gas - p_water, but at cells truncated by SWL, the code
+    # will give p_oil - p_water. Delete these inconsistent capillary
+    # pressures by ignoring PC scaling whenever SWL has been truncated:
+    if "QC_FLAG" in grid_df:
+        grid_df.loc[grid_df["QC_FLAG"] == __SWL_TRUNC__, "PC_SCALING"] = np.nan
+
+    if "PC_SCALING" in grid_df:
+        grid_df["PC"] = compute_pc(grid_df, satfunc_df)
+
+    return grid_df
+
+
+def qc_flag(qc_frame):
+    """Compute a series categorizing the QC type of the cell, determining
+    how SWATINIT behaved in that cell
+
+    This function is the core of check_swatinit.
+
+    Args:
+        qc_frame (pd.DataFrame)
+
+    Returns:
+        pd.DataFrame (with an additional column QC_FLAG)
+    """
+
+    qc_col = pd.Series(index=qc_frame.index, dtype=str)
+
+    # SWATINIT ignored, water is lost if pc > 0, lost and/or gained if oil-wet pc-curve
+    qc_col[
+        (qc_frame["SWATINIT"] < 1) & (qc_frame["Z"] > qc_frame["OWC"])
+    ] = __HC_BELOW_FWL__
+
+    # SWATINIT accepted and PC is scaled.
+    qc_col[
+        np.isclose(qc_frame["SWAT"], qc_frame["SWATINIT"], atol=1e-6)
+        & (qc_frame["SWATINIT"] < 1)
+    ] = __PC_SCALED__
+
+    if "PC_SCALING" in qc_frame:
+        # If SWATINIT == SWL and capillary pressure warrants SWL, then PPCW
+        # matches PCOW_MAX and PC_SCALING is 1. We denote this as __PC_SCALED__
+        # because it is "scaled" by 1.
+        qc_col[
+            np.isclose(qc_frame["SWAT"], qc_frame["SWATINIT"], atol=1e-6)
+            & np.isclose(qc_frame["PC_SCALING"], 1)
+        ] = __PC_SCALED__
+
+    # PC is scaled (including "scaling" using PC_SCALING=1), but
+    # SWAT!=SWATINIT, this must be due to EQUIL item 9 being nonzero.
+    if "OIP_INIT" in qc_frame and "PC_SCALING" in qc_frame:
+        qc_col[
+            (~np.isclose(qc_frame["OIP_INIT"], 0))
+            & (~np.isclose(qc_frame["SWAT"], qc_frame["SWATINIT"], atol=1e-6))
+            & (~pd.isnull(qc_frame["PC_SCALING"]))
+        ] = __EQUIL__
```

Comment:
__EQUIL__ might not be the most descriptive label... Perhaps '__FINEGRID_EQUIL__' , '__FINE_EQUIL__' or some better suggestion?

Reply:
The label the user will see is currently `EQUIL_INIT`. But `FINE_EQUIL` possibly sounds better.